test/utils/server.js

```diff
@@ -1,73 +1,43 @@
-    conf: yaml.safeLoad(fs.readFileSync(`${__dirname}/../../config.yaml`))
-};
-// build the API endpoint URI by supposing the actual service
-// is the last one in the 'services' list in the config file
-const myServiceIdx = config.conf.services.length - 1;
-const myService = config.conf.services[myServiceIdx];
-config.uri = `http://localhost:${myService.conf.port}/`;
-config.service = myService;
-// no forking, run just one process when testing
-config.conf.num_workers = 0;
-// have a separate, in-memory logger only
-config.conf.logging = {
-    name: 'test-log',
-    level: 'trace',
-    stream: logStream()
-};
-// make a deep copy of it for later reference
-const origConfig = extend(true, {}, config);
-
-module.exports.stop = () => { return BBPromise.resolve(); };
-let options = null;
-const runner = new ServiceRunner();
-
-
-function start(_options) {
-
-    _options = _options || {};
-
-    if (!assert.isDeepEqual(options, _options)) {
-        console.log('starting test server'); // eslint-disable-line no-console
-        return module.exports.stop().then(() => {
-            options = _options;
-            // set up the config
-            config = extend(true, {}, origConfig);
-            extend(true, config.conf.services[myServiceIdx].conf, options);
-            return runner.start(config.conf)
-            .then((serviceReturns) => {
-                module.exports.stop = () => {
-                    console.log('stopping test server'); // eslint-disable-line no-console
-                    serviceReturns.forEach(servers =>
-                        servers.forEach(server =>
-                            server.shutdown()));
-                    return runner.stop().then(() => {
-                        module.exports.stop = () => {
-                            return BBPromise.resolve();
-                        };
-                    });
-                };
-                return true;
-            });
-        });
-    } else {
-        return BBPromise.resolve();
+    get config() {
+        if (!this._running) {
+            throw new Error('Accessing test service config before starting the service');
+        }
+
+        // build the API endpoint URI by supposing the actual service
+        // is the last one in the 'services' list in the config file
+        const myServiceIdx = this._runner._impl.config.services.length - 1;
+        const myService = this._runner._impl.config.services[myServiceIdx];
+        const uri = `http://localhost:${myService.conf.port}/`;
+        if (!this._spec) {
+            // We only want to load this once.
+            const specPath = `${__dirname}/../../../${myService.spec ? myService.spec : 'spec.yaml'}`;
+            try {
+                this._spec = yaml.safeLoad(fs.readFileSync(specPath));
+            } catch (e) {
+                // this error will be detected later, so ignore it
+                this._spec = { paths: {}, 'x-default-params': {} };
+            }
+        }
```

Comment:
Let's not get the spec like this. Why is it needed at all? Where it's needed we need to pull it async from an endpoint.

Reply:
Changed in latest commit. The config property now queries the API for the spec.